src/odemis/acq/test/fastem_test.py

```diff
@@ -437,40 +428,30 @@ def on_progress_update(self, future, start, end):
 
 
 class TestFastEMAcquisitionTask(unittest.TestCase):
-    """Test methods of the fastem.AcquistionTask class."""
+    """Test methods of the fastem.AcquisitionTask class."""
 
     @classmethod
     def setUpClass(cls):
         if TEST_NOHW:
-            # If we are testing without hardware we just need a few attributes to be set correctly.
-            cls.scanner = None
-            cls.asm = None
-
-            # Use Mocks of the classes to be able to call the fake VAs as for instance mppc.dataContent.value
-            cls.mppc = Mock()
-            cls.mppc.dataContent.value = 'empty'
-
-            cls.multibeam = Mock()
-            cls.multibeam.pixelSize.value = (4.0e-9, 4.0e-9)
-            cls.multibeam.resolution.value = (800, 800)
-
-            cls.descanner = None
-            cls.stage = None
-            cls.ccd = None
-            cls.beamshift = None
-            cls.lens = None
+            test.start_backend(FASTEM_CONFIG_ASM)
         else:
-            # get the hardware components from the simulators or hardware
-            cls.scanner = model.getComponent(role='e-beam')
-            cls.asm = model.getComponent(role="asm")
-            cls.mppc = model.getComponent(role="mppc")
-            cls.multibeam = model.getComponent(role="multibeam")
-            cls.descanner = model.getComponent(role="descanner")
-            cls.stage = model.getComponent(
-                role="stage")  # TODO replace with stage-scan when ROA conversion method available
-            cls.ccd = model.getComponent(role="diagnostic-ccd")
-            cls.beamshift = model.getComponent(role="ebeam-shift")
-            cls.lens = model.getComponent(role="lens")
+            raise unittest.SkipTest("No hardware backend file available, skipping tests.")
+
+        # Get the hardware components from the simulators or hardware
+        cls.scanner = model.getComponent(role='e-beam')
+        cls.asm = model.getComponent(role="asm")
+        cls.mppc = model.getComponent(role="mppc")
+        cls.multibeam = model.getComponent(role="multibeam")
+        cls.descanner = model.getComponent(role="descanner")
+        cls.stage = model.getComponent(
+            role="stage")  # TODO replace with stage-scan when ROA conversion method available
+        cls.ccd = model.getComponent(role="diagnostic-ccd")
+        cls.beamshift = model.getComponent(role="ebeam-shift")
+        cls.lens = model.getComponent(role="lens")
+
+        cls.beamshift.updateMetadata({model.MD_CALIB: cls.scanner.beamShiftTransformationMatrix.value})
```

Comment:
Please add a comment why you have to do it "manually". Where is it done normally?